tests/tests/bitasset_tests.cpp

```diff
@@ -542,6 +544,303 @@ BOOST_AUTO_TEST_CASE( hf_890_test )
+   trx.set_expiration( HARDFORK_CORE_922_931_TIME + global_params.maintenance_interval + global_params.maximum_time_until_expiration );
+
+   ACTORS( (nathan) (john) );
+
+   assets_922_931 asset_objs = create_assets_922_931( this );
+   const asset_id_type& bit_usd_id = asset_objs.bit_usd->get_id();
+
+   // make a generic operation
+   bitasset_evaluator_wrapper evaluator;
+   evaluator.set_db( db );
+   asset_update_bitasset_operation op;
+   op.asset_to_update = bit_usd_id;
+   op.issuer = asset_objs.bit_usd->issuer;
+   op.new_options = asset_objs.bit_usd->bitasset_data(db).options;
+
+   // this should pass
+   BOOST_TEST_MESSAGE( "Evaluating a good operation" );
+   BOOST_CHECK( evaluator.evaluate(op) == void_result() );
+
+   // test with a market issued asset
+   BOOST_TEST_MESSAGE( "Sending a non-bitasset." );
+   op.asset_to_update = asset_objs.user_issued->get_id();
+   REQUIRE_EXCEPTION_WITH_TEXT( evaluator.evaluate(op), "Cannot update BitAsset-specific settings on a non-BitAsset" );
+   op.asset_to_update = bit_usd_id;
+
+   // test changing issuer
+   BOOST_TEST_MESSAGE( "Test changing issuer." );
+   account_id_type original_issuer = op.issuer;
+   op.issuer = john_id;
+   REQUIRE_EXCEPTION_WITH_TEXT( evaluator.evaluate(op), "Only asset issuer can update" );
+   op.issuer = original_issuer;
+
+   // bad backing_asset
+   BOOST_TEST_MESSAGE( "Non-existent backing asset." );
+   asset_id_type correct_asset_id = op.new_options.short_backing_asset;
+   op.new_options.short_backing_asset = asset_id_type();
+   op.new_options.short_backing_asset.instance = 123;
+   REQUIRE_EXCEPTION_WITH_TEXT( evaluator.evaluate(op), "Unable to find" );
+   op.new_options.short_backing_asset = correct_asset_id;
+
+   // now check the things that are wrong and won't pass after HF 922 / 931
+   BOOST_TEST_MESSAGE( "Now check the things that are wrong and won't pass after HF 922 / 931" );
+
+   // back by self
+   BOOST_TEST_MESSAGE( "Back by itself" );
+   op.new_options.short_backing_asset = bit_usd_id;
+   REQUIRE_EXCEPTION_WITH_TEXT( evaluator.evaluate(op), "Cannot update an asset to be backed by itself" );
+   op.new_options.short_backing_asset = correct_asset_id;
+
+   // prediction market with different precision
+   BOOST_TEST_MESSAGE( "Prediction market with different precision" );
+   op.asset_to_update = asset_objs.prediction->get_id();
+   op.issuer = asset_objs.prediction->issuer;
+   REQUIRE_EXCEPTION_WITH_TEXT( evaluator.evaluate(op), "The precision of the asset and backing asset must" );
+   op.asset_to_update = bit_usd_id;
+   op.issuer = asset_objs.bit_usd->issuer;
+
+   // checking old backing asset instead of new backing asset
+   BOOST_TEST_MESSAGE( "Correctly checking new backing asset rather than old backing asset" );
+   op.new_options.short_backing_asset = asset_objs.bit_usdbacked->get_id();
+   REQUIRE_EXCEPTION_WITH_TEXT( evaluator.evaluate(op), "May not modify a blockchain-controlled market asset to" );
```

Comment:
Please change the expected message to be more accurate, "May not modify a blockchain-controlled market asset to" occurred several times in the code.

Reply:
Done